app/controllers/api/connections_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::ConnectionsController < ApplicationController
+    include AuthenticationConcern
+
+    def request_match
+
+        if @current_user
+
+            request = MatchRequest.new
+            request.student_id = @current_user.id
+
+            if SkillClass.find_by(id: params[:id])
+                #check if class exists
+                request.class_id = params[:id]
```

Comment:
Ok, I understand that by now you there are already almost as many comments in this code review as lines of code, but if you follow the suggestion to merge lines 8-9 to end up with `request = MatchRequest.new(student_id: @current_user.id)` **and** if you move the validation from line 11 to a `before_action` block, you can also merge this line into the line 8 initializer and end up with `request = MatchRequest.new(student_id: @current_user.id, class_id: params[:id])`